### DISCUSSION TITLE
Best way to customize shop pages

### DISCUSSION THREAD

**Discussion by tomwesterling** [bootscore/bootscore#537] [Q&A] [answered]
Hi there,

I am really enjoying to build my new project based on the bootscore theme. However, right now I am having some difficulties / lackage in knowledge when it comes to customizing the shop pages. For example, I want to build a custom shop page. By default, the grid of WooCommerce products shows up. Let's say I want to insert an image and some text on top of the page. Also, I don't want to show all the products at once, but have a section that says "tickets", followed by products in that speficic cateogry.
Similar applies for the single product page and the product (category) archive pages.

I have noticed, that overriding WooCommerce's core templates is not the best idea, since I would need to delete bootscore's "woocommerce.php" file. Maybe I am wrong and this is the intended way of customizing the template.

I want to do it the right and clean way, so If you have any advice on how to customize shop pages using bootscore I'm very thankful.! :)

Best,

Tom

**Comment by crftwrk** [comment, marked as answer]
> I have noticed, that overriding WooCommerce's core templates is not the best idea, since I would need to delete bootscore's "woocommerce.php" file. Maybe I am wrong and this is the intended way of customizing the template.

Yes, `woocommerce.php` is used to get the breadcrumb right in place, filters won't work that way. So using this file is the best option even it doesn't allow to customise the shop pages.

You can add an image and text/custom HTML to the category. But the best way is to create a new page, design whatever you want and use shortcodes `[products]` which are included https://woocommerce.com/document/woocommerce-shortcodes/#products.

Solved?

**Reply by tomwesterling** [reply]
Thank you for the fast response!

I was also thinking about your suggestion to create a blank custom page and fill it with shortcodes as needed.

Instead of creating a new page via the WP dashboard and edit it with a site editor like Gutenberg, I would prefer to create a file like "shop.php" in my child theme's root directory. This gives me maximum flexibility and comfort to work inside VS Code.

Now I am wondering how to link this custom php file to the WooCommerce shop page, since WP pages are usually no individual php files but stored in a database.

Possibly a redirect would work, but I guess there's better solutions?

Again, thanks for the support!

Tom


==== 

Edit: I found out that this can be done by the following steps:
1. Create "shop.php" in child theme's root directory
2. Insert the following code at the top of the file:

`<?php
/*
Template Name: MyShopTemplate
*/
?>
`

3. Add a new page via the WP dashboard
4. On the right bar, select "MyShopTemplate" under the "Template" setting.
5. Start editing "shop.php" in whatever IDE of choice.

**Reply by crftwrk** [reply]
You can also try to delete `woocommerce.php`and using hooks instead https://woocommerce.com/document/woocommerce-theme-developer-handbook/#section-3. If it works fine, we can merge it here in main-theme.
